src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -990,6 +991,27 @@ private void UpdateNextRefreshTime(KeyValueWatcher changeWatcher)
             Func<ConfigurationClient, Task<T>> funcToExecute,
             CancellationToken cancellationToken = default)
         {
+            if (_options.LoadBalancingEnabled && _lastSuccessfulEndpoint != null)
```

Comment:
It seems to also make sense to check if there is more than one client in `clients`